domains/client/domain-executor/src/tests.rs

```diff
@@ -68,12 +69,6 @@ async fn test_executor_full_node_catching_up() {
     // Bob is able to sync blocks.
     futures::future::join(alice.wait_for_blocks(3), bob.wait_for_blocks(3)).await;
 
-    assert_eq!(
-        ferdie.client.info().best_number,
-        alice.client.info().best_number,
-        "Primary chain and system domain must be on the same best height"
-    );
```

Comment:
The test `test_executor_full_node_catching_up` is also flakey due to this check. This assertion is incorrect as the primary chain node can always produce new blocks while the system domain node is syncing the previous blocks and does not sync the newest block in time.

And I can also see that in the CI log, the primary chain has produced the fourth block while the system domain is syncing the first 3 blocks, thus removing this assertion.